json_encoder.go

```diff
@@ -151,24 +156,32 @@ func (enc *jsonEncoder) AddFields(fields []Field) {
 	addFields(enc, fields)
 }
 
-// WriteMessage writes a complete log message to the supplied writer, including
+// WriteEntry writes a complete log message to the supplied writer, including
 // the encoder's accumulated fields. It doesn't modify or lock the encoder's
 // underlying byte slice. It's safe to call from multiple goroutines, but it's
-// not safe to call CreateMessage while adding fields.
-func (enc *jsonEncoder) WriteMessage(sink io.Writer, lvl string, msg string, ts time.Time) error {
+// not safe to call WriteEntry while adding fields.
+func (enc *jsonEncoder) WriteEntry(sink io.Writer, e *Entry) error {
+	if sink == nil {
+		return errNilSink
+	}
+	if e == nil {
+		return errNilEntry
+	}
 	// Grab an encoder from the pool so that we can re-use the underlying
 	// buffer.
 	final := newJSONEncoder()
 	defer final.Free()
 
 	final.bytes = append(final.bytes, `{"msg":"`...)
-	final.safeAddString(msg)
+	final.safeAddString(e.Message)
 	final.bytes = append(final.bytes, `","level":"`...)
-	final.bytes = append(final.bytes, lvl...)
+	final.bytes = append(final.bytes, e.Level.String()...)
 	final.bytes = append(final.bytes, `","ts":`...)
-	final.bytes = strconv.AppendInt(final.bytes, ts.UnixNano(), 10)
+	final.bytes = strconv.AppendFloat(final.bytes, timeToSeconds(e.Time), 'f', -1, 64)
```

Comment:
Let's add a test for writing out a future time (maybe +100 years) and ensure that it's written without exponents